presto-spark-base/src/main/java/com/facebook/presto/spark/execution/operator/NativeExecutionOperator.java

```diff
@@ -187,9 +187,13 @@ public Page getOutput()
 
             return null;
         }
-        catch (InterruptedException e) {
+        catch (InterruptedException | RuntimeException e) {
+            String error = e.getMessage();
+            if (!process.isAlive()) {
+                error = String.format("Native process has crashed. %s", e.getMessage());
+            }
             log.error(e);
-            throw new RuntimeException(e);
+            throw new RuntimeException(error, e);
```

Comment:
Why do we want to re-wrap a RuntimeException here? Should we re-throw instead?

Reply:
Should we use `PrestoException(GENERIC_INTERNAL_ERROR, ...)` here instead?